lib/components/article_progress_tracker.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+import 'package:maui/db/entity/user.dart';
+import 'package:maui/repos/article_progress_repo.dart';
+import 'package:maui/state/app_state_container.dart';
+
+class ArticleProgressTracker extends StatefulWidget {
+  final String topicId;
+
+  ArticleProgressTracker({Key key, this.topicId}) : super(key: key);
+
+  @override
+  State createState() => new ArticleProgressTrackerState();
+}
+
+class ArticleProgressTrackerState extends State<ArticleProgressTracker> {
+  double _articleProgress;
+  bool _isLoading = true;
+  User user;
+
+  void _initData() async {
+    super.initState();
+    articleProgressTracker();
+    setState(() => _isLoading = false);
+  }
+
+   void articleProgressTracker() async {
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      user = AppStateContainer.of(context).state.loggedInUser;
+      _articleProgress = await ArticleProgressRepo()
+        .getArticleProgressStatus(widget.topicId, user.id);
+    });
+    setState(() => _isLoading = false);
+  }
+
+  
+
+  @override
+    void didChangeDependencies() {
+      // TODO: implement didChangeDependencies
+      super.didChangeDependencies();
+      _initData();
```

Comment:
Better to use addPostFrameCallback (as you have used) rather than didChangeDependencies

Reply:
No need for this since you are using addPostFrameCallback